src/core/importType.js

```diff
@@ -21,10 +21,11 @@ export function isAbsolute(name) {
   return name.indexOf('/') === 0
 }
 
-export function isBuiltIn(name, settings) {
+// path is defined only when a resolver resolves to a non-standard path
+export function isBuiltIn(name, settings, path) {
   const base = baseModule(name)
   const extras = (settings && settings['import/core-modules']) || []
-  return coreModules[base] || extras.indexOf(base) > -1
+  return !path && coreModules[base] || extras.indexOf(base) > -1
```

Comment:
let's change this to an early return (at the top of the function, ideally):
```suggestion
  if (!path) return false
  return coreModules[base] || extras.indexOf(base) > -1
```